test/test_crypto.py

```diff
@@ -0,0 +1,169 @@
+
+rsa_private_key_bytes = b"\xeb\xa5wc1\x1c\xd2\xa4\xdd\xaaT\xf3kb%\xfd\xe3?^\x84\x12\xb6+P,\xbc\xa4g\xff\xb2=` \x0e\xbd\x17\\\x0b`j\xd5\\\xb4U:\xd6\xc7\xa3^\xc7|\xcc\x1cn\x06M\x8a\xad\x9d\x9dn\xb9\xf3\xc6\xcc\xf2\xb6 6\xfa\x91s\xb1\x8f \xe1|\x00\xe7\xee\xd9\xb3\xcc\x91\xac\xc0\x08K\xa8\xcd\xba\xe3\x81\x8e\xc7\x8a\xf0\xf7\xee\x11\xa3\xbc\xc0\xd6\xc9\x19\x8ch\x10\x9bN\xbc\x19u3\xf59\x1e\xb7\x9a&\xc6\xa1\x9e{\x85\x98\r"
+
+ecc_private_key = b"""
+-----BEGIN EC PRIVATE KEY-----
+MHcCAQEEIMJI9ujmlT/qftbXWlMwOSpkxiWLAbyIMWEFPOqTbXYMoAoGCCqGSM49
+AwEHoUQDQgAEgO/tHxp/YOuP4wAV3w66C8JNiSHOKSAYtlNKSN4ZDI//wn0f7zBv
+Uc7FqaRPA9LL6k6C1YfdOi/yvTB7Y4Tgaw==
+-----END EC PRIVATE KEY-----
+"""
+
+ecc_public_key = b"""-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEgO/tHxp/YOuP4wAV3w66C8JNiSHO
+KSAYtlNKSN4ZDI//wn0f7zBvUc7FqaRPA9LL6k6C1YfdOi/yvTB7Y4Tgaw==
+-----END PUBLIC KEY-----
+"""
+
+ecc_public_key_bytes = b"\x80\xef\xed\x1f\x1a\x7f`\xeb\x8f\xe3\x00\x15\xdf\x0e\xba\x0b\xc2M\x89!\xce) \x18\xb6SJH\xde\x19\x0c\x8f\xff\xc2}\x1f\xef0oQ\xce\xc5\xa9\xa4O\x03\xd2\xcb\xeaN\x82\xd5\x87\xdd:/\xf2\xbd0{c\x84\xe0k"
+
+ecc_private_key_bytes = b"\xc2H\xf6\xe8\xe6\x95?\xea~\xd6\xd7ZS09*d\xc6%\x8b\x01\xbc\x881a\x05<\xea\x93mv\x0c"
+
+
+class CryptoTest(TSS2_EsapiTest):
+    def test_public_from_pem_rsa(self):
+        pub = types.TPM2B_PUBLIC()
+        crypto.public_from_pem(rsa_public_key, pub)
+
+        self.assertEqual(pub.publicArea.type, types.TPM2_ALG.RSA)
+        self.assertEqual(pub.publicArea.parameters.rsaDetail.keyBits, 2048)
+        self.assertEqual(pub.publicArea.parameters.rsaDetail.exponent, 0)
+        self.assertEqual(bytes(pub.publicArea.unique.rsa.buffer), rsa_public_key_bytes)
+
+    def test_private_from_pem_rsa(self):
+        priv = types.TPM2B_SENSITIVE()
+        crypto.private_from_pem(rsa_private_key, priv)
+
+        self.assertEqual(priv.sensitiveArea.sensitiveType, types.TPM2_ALG.RSA)
+        self.assertEqual(
+            bytes(priv.sensitiveArea.sensitive.rsa.buffer), rsa_private_key_bytes
+        )
+
+    def test_loadexternal_rsa(self):
+        pub = types.TPM2B_PUBLIC.fromPEM(rsa_public_key)
+        pub.publicArea.nameAlg = TPM2_ALG.SHA256
+        pub.publicArea.objectAttributes = (
+            TPMA_OBJECT.DECRYPT | TPMA_OBJECT.SIGN_ENCRYPT | TPMA_OBJECT.USERWITHAUTH
+        )
+        pub.publicArea.parameters.rsaDetail.symmetric.algorithm = TPM2_ALG.NULL
+        pub.publicArea.parameters.rsaDetail.scheme.scheme = TPM2_ALG.NULL
```

Comment:
I think fromPEM should include the objectAttributes and the scheme, we can default them to what you have here, or possibly force folks to specify them so we don't hand them back a partially constructed object. Thoughts?

Reply:
Sounds good, that + setting nameAlg to SHA256 should make the key ready to be used directly